File: SRP/Example/MarsRover/MarsRoverKata/Commands/TurnLeftCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MarsRoverKata.Commands
{
    public class TurnLeftCommand : ICommand
    {
        public Rover Rover { get; set; }

        public TurnLeftCommand(Rover rover)
        {
            Rover = rover;
        }

        public bool Execute()
        {
            return Rover.TurnLeft();
        }
    }
}

File: LSP+ISP/Java/MarsRoverKata/src/main/java/MarsRoverKata/Commands/TurnLeftCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.Rover;

public class TurnLeftCommand implements ICommand {
    private Rover rover;

    public Rover getRover() {
        return rover;
    }

    public void setRover(Rover rover) {
        this.rover = rover;
    }

    public TurnLeftCommand(Rover rover) {
        this.rover = rover;
    }

    public boolean execute() {
        return rover.turnLeft();
    }
}

